src/handler.rs

```diff
@@ -38,6 +38,22 @@ pub fn handle_key_events(
                 app.quit();
             }
         }
+        KeyCode::Char(' ') => {
+            if app.show_details {
+                let references = &app.list.selected().unwrap().references;
+                // Iterates over the references and find the first http reference, if any
+                if let Some(first_http_reference) =
+                    references.iter().find(|r| r.starts_with("http"))
+                {
+                    match webbrowser::open(first_http_reference) {
+                        Ok(_) => {} // Opened the browser successfully, nothing to do
+                        Err(err) => {
+                            println!("Failed to open browser: {err:?}");
+                        }
+                    }
+                }
+            }
```

Comment:
We can rewrite this in a more compact and simple way like this:

```suggestion
                if let Some(reference) = app
                    .list
                    .selected()
                    .map(|v| &v.references)
                    .and_then(|references| references.iter().find(|r| r.starts_with("http")))
                {
                    if let Err(e) = webbrowser::open(reference) {
                        eprintln!("Failed to open browser: {e:?}");
                    }
                }
```

I made the following changes:
1. Use `map` to get the references from the selected item. This will avoid `unwrap` (i.e. panicking).
2. If we are able to get the references, find the first one that starts with "http".
3. If all the conditions are met so far, open via web browser and print error if any. Use `eprintln` instead of `println` for error-printing. Also, hint #3